Please consider renaming function airflow.utils.dag_cycle_tester.test_cycle to check_cycle

**Description**

function test_cycle will be collected by pytest if this function is imported in a user-defined test.
Such an error will be emitted as pytest will treat it as a test.
```
test setup failed
file /opt/conda/lib/python3.8/site-packages/airflow/utils/dag_cycle_tester.py, line 27
  def test_cycle(dag):
E       fixture 'dag' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

```

**Use case / motivation**
This can be simply "fixed" by renaming this function to check_cycle although the user can do it upon importing this function.
This change by airflow will reduce the amount of surprise received by the end-user, and save a large number of debugging hours. 

**Are you willing to submit a PR?**
The changeset is too small to worth the trouble for a PR.

--- a/airflow/models/dagbag.py
+++ b/airflow/models/dagbag.py
@@ -44,7 +44,7 @@ from airflow.exceptions import (
 )
 from airflow.stats import Stats
 from airflow.utils import timezone
-from airflow.utils.dag_cycle_tester import test_cycle
+from airflow.utils.dag_cycle_tester import check_cycle
 from airflow.utils.file import correct_maybe_zipped, list_py_file_paths, might_contain_dag
 from airflow.utils.log.logging_mixin import LoggingMixin
 from airflow.utils.retries import MAX_DB_RETRIES, run_with_db_retries
@@ -427,7 +427,7 @@ class DagBag(LoggingMixin):
         The only purpose of this is to avoid exposing ``recursive`` in ``bag_dag()``,
         intended to only be used by the ``_bag_dag()`` implementation.
         """
-        test_cycle(dag)  # throws if a task cycle is found
+        check_cycle(dag)  # throws if a task cycle is found
 
         dag.resolve_template_files()
         dag.last_loaded = timezone.utcnow()

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -31,6 +31,7 @@ def test_cycle(dag):
     will be considered as a unit test by pytest, resulting in failure.
     """
     from warnings import warn
+
     warn(
         "Deprecated, please use `check_cycle` at the same module instead.",
         DeprecationWarning,

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -27,7 +27,7 @@ CYCLE_DONE = 2
 def test_cycle(dag):
     """
     A wrapper function of `check_cycle` for backward compatibility purpose.
-    New code should use `check_cycle` instead since the function `test_cycle` starts with test_ and
+    New code should use `check_cycle` instead since this function name `test_cycle` starts with 'test_' and
     will be considered as a unit test by pytest, resulting in failure.
     """
     from warnings import warn

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -26,9 +26,9 @@ CYCLE_DONE = 2
 
 def test_cycle(dag):
     """
-    A wrapper function of check_cycle for backward compatibility reason.
-    New code should use check_cycle instead. Since the function test_cycle starts with test_, pytest
-    will consider it as a unit test and cause failure.
+    A wrapper function of `check_cycle` for backward compatibility purpose.
+    New code should use `check_cycle` instead since the function `test_cycle` starts with test_ and
+    will be considered as a unit test by pytest, resulting in failure.
     """
     from warnings import warn
     warn("Deprecated, please use `check_cycle` at the same module instead.", DeprecationWarning)

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -31,7 +31,11 @@ def test_cycle(dag):
     will be considered as a unit test by pytest, resulting in failure.
     """
     from warnings import warn
-    warn("Deprecated, please use `check_cycle` at the same module instead.", DeprecationWarning)
+    warn(
+        "Deprecated, please use `check_cycle` at the same module instead.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
     return check_cycle(dag)
 
 

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -27,8 +27,8 @@ CYCLE_DONE = 2
 def test_cycle(dag):
     """
     A wrapper function of check_cycle for backward compatibility reason.
-    New code should use check_cycle instead, since this function name test_cycle starts with test_ and pytest
-    will consider it as a unit test causing failure.
+    New code should use check_cycle instead. Since the function test_cycle starts with test_, pytest
+    will consider it as a unit test and cause failure.
     """
     from warnings import warn
     warn("Deprecated, please use check_cycle at the same module instead.", DeprecationWarning)

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -26,6 +26,17 @@ CYCLE_DONE = 2
 
 def test_cycle(dag):
     """
+    A wrapper function of check_cycle for backward compatibility reason.
+    New code should use check_cycle instead, since this function name test_cycle starts with test_ and pytest
+    will consider it as a unit test causing failure.
+    """
+    from warnings import warn
+    warn("Deprecated, please use check_cycle at the same module instead.", DeprecationWarning)
+    return check_cycle(dag)
+
+
+def check_cycle(dag):
+    """
     Check to see if there are any cycles in the DAG. Returns False if no cycle found,
     otherwise raises exception.
     """

--- a/airflow/utils/dag_cycle_tester.py
+++ b/airflow/utils/dag_cycle_tester.py
@@ -31,7 +31,7 @@ def test_cycle(dag):
     will consider it as a unit test and cause failure.
     """
     from warnings import warn
-    warn("Deprecated, please use check_cycle at the same module instead.", DeprecationWarning)
+    warn("Deprecated, please use `check_cycle` at the same module instead.", DeprecationWarning)
     return check_cycle(dag)
 
 

--- a/tests/utils/test_dag_cycle.py
+++ b/tests/utils/test_dag_cycle.py
@@ -22,7 +22,7 @@ import pytest
 from airflow import DAG
 from airflow.exceptions import AirflowDagCycleException
 from airflow.operators.dummy import DummyOperator
-from airflow.utils.dag_cycle_tester import test_cycle as _test_cycle  # to avoid being a test function
+from airflow.utils.dag_cycle_tester import check_cycle
 from tests.models import DEFAULT_DATE
 
 
@@ -31,7 +31,7 @@ class TestCycleTester(unittest.TestCase):
         # test empty
         dag = DAG('dag', start_date=DEFAULT_DATE, default_args={'owner': 'owner1'})
 
-        assert not _test_cycle(dag)
+        assert not check_cycle(dag)
 
     def test_cycle_single_task(self):
         # test single task
@@ -40,7 +40,7 @@ class TestCycleTester(unittest.TestCase):
         with dag:
             DummyOperator(task_id='A')
 
-        assert not _test_cycle(dag)
+        assert not check_cycle(dag)
 
     def test_semi_complex(self):
         dag = DAG('dag', start_date=DEFAULT_DATE, default_args={'owner': 'owner1'})
@@ -77,7 +77,7 @@ class TestCycleTester(unittest.TestCase):
             op2.set_downstream(op4)
             op5.set_downstream(op6)
 
-        assert not _test_cycle(dag)
+        assert not check_cycle(dag)
 
     def test_cycle_loop(self):
         # test self loop
@@ -89,7 +89,7 @@ class TestCycleTester(unittest.TestCase):
             op1.set_downstream(op1)
 
         with pytest.raises(AirflowDagCycleException):
-            assert not _test_cycle(dag)
+            assert not check_cycle(dag)
 
     def test_cycle_downstream_loop(self):
         # test downstream self loop
@@ -109,7 +109,7 @@ class TestCycleTester(unittest.TestCase):
             op5.set_downstream(op5)
 
         with pytest.raises(AirflowDagCycleException):
-            assert not _test_cycle(dag)
+            assert not check_cycle(dag)
 
     def test_cycle_large_loop(self):
         # large loop
@@ -127,7 +127,7 @@ class TestCycleTester(unittest.TestCase):
 
             current.set_downstream(start)
         with pytest.raises(AirflowDagCycleException):
-            assert not _test_cycle(dag)
+            assert not check_cycle(dag)
 
     def test_cycle_arbitrary_loop(self):
         # test arbitrary loop
@@ -149,4 +149,4 @@ class TestCycleTester(unittest.TestCase):
             op5.set_downstream(op1)
 
         with pytest.raises(AirflowDagCycleException):
-            assert not _test_cycle(dag)
+            assert not check_cycle(dag)

